flink-dist/src/main/flink-bin/mesos-bin/mesos-appmaster.sh

```diff
@@ -38,9 +38,14 @@ constructAppMasterClassPath() {
     echo $CC_CLASSPATH:$INTERNAL_HADOOP_CLASSPATHS
 }
 
+if [ "$FLINK_IDENT_STRING" = "" ]; then
+    FLINK_IDENT_STRING="$USER"
+fi
+
+
 CC_CLASSPATH=`manglePathList $(constructAppMasterClassPath)`
 
-log=flink-appmaster.log
+log="${FLINK_LOG_DIR}/flink-${FLINK_IDENT_STRING}-mesos-appmaster-${HOSTNAME}.log"
```

Comment:
Thanks for clarifying this. I think in the future it could make sense to unify the different deployment scripts and fix these issues.